site/blog/harder-better-prompter-stronger.md

```diff
@@ -0,0 +1,345 @@
+
+## Why system prompt hardening matters
+
+We (unwillingly) accept that users will attempt to mess with AI systems. Let's reveal (_insert cape flourish here_) some of the consequences:
+
+- The LLM tone drifts
+- Roleplaying as other personas
+- Safety filters ignored
+- Reverse engineering for prompt injection refinement
+- Impersonated (brand, users - anything)
+- Exposing sensitive data
+- Enabling developer-only modes (!!)
+
+## Techniques to harden system prompts
+
+There are multiple strategies to harden system prompts. To name a few:
+
+1. **Instruction shielding**: Prevent the model from accepting new instructions that override the original intended behaviour.
+2. **Syntax reinforcement**: Make a prompt's structure harder to manipulate.
+3. **Layered prompting**: Use multiple models or prompt stages.
+
+Ultimately, we should have a pipeline for our prompts consisting of tools. Most of the tooling and techniques (like guards, prompt chaining, etc.) operate outside the LLM; prompt hardening happens at the LLM level.
+
+## How to harden a system prompt
+
+Imagine we've started building our prompt pipeline. We're excited to send out our app into the wild. It's a tad uncomfortable because we don't have anything to confirm our system prompt's strength - so it's time for evaluations (evals).
+
+We can do this with Promptfoo (I work there - _>what a coincidence_). It's open source, so you can get started yourself. I assure you that I have not accidentally uploaded any API keys, so you'd need your own.
+
+```yaml title="promptfooconfig.yaml"
+prompts:
+  - 'Convert this English to {{language}}: {{input}}'
+  - 'Translate to {{language}}: {{input}}'
+  - file://path/to/prompt.txt
+
+providers:
+  - 'openai:gpt-4o'
+  - 'anthropic:claude-3-5-sonnet'
+
+defaultTest:
+  assert:
+    - type: not-contains
+      value: AI language model
+
+tests:
+  - vars:
+      language: French
+      input: Hello world
+    assert:
+      - type: contains
+        value: 'Bonjour le monde'
+  - vars:
+      language: Spanish
+      input: Where is the library?
+    assert:
+      - type: icontains
+        value: 'Dónde está la biblioteca'
+```
+
+There are four main parts to the file:
+
+- `prompts` - The prompts to test, such as our unhardened system prompts.
+- `providers` - The LLM providers and configuration.
+- `defaultTest` - Default test properties applied to all tests.
+- `tests` - Individual tests. Note the `{{variables}}` from the prompts showing up under vars.
+
+There's a handy table for [assertion types](https://www.promptfoo.dev/docs/configuration/expected-outputs/#assertion-types) if you're interested.
+
+For our demonstration, we'll be skipping the defaultTest section, which is optional. Assertions are also technically optional, but we'll be using them to demonstrate the effects of system prompt hardening.
```

Comment:
Maybe we can just remove `defaultTest` from the original example to make it less intimidating and/or confusing

Reply:
If you want me to remove it from the original getting started page lmk - I have removed it from the example here